File: problems/573/solution.cpp
class Solution {
public:
    int minDistance(int height, int width, vector<int>& tree, vector<int>& squirrel, vector<vector<int>>& nuts) {
        int ans = 0, savedDistance = INT_MIN, two;
        
        for (const vector<int>& nut: nuts) {
            ans += 2 * dist(tree, nut);
            
            // one = max(one, dist(nut, tree) - dist(squirrel, nut));
            if (dist(tree, nut) - dist(nut, squirrel) > savedDistance) {
                savedDistance = dist(tree, nut) - dist(nut, squirrel);
                two = -dist(tree, nut) + dist(nut, squirrel);
            }
        }
        
        return ans + two;
    }
    
    int dist(const vector<int>& a, const vector<int>& b) {
        return abs(a[0] - b[0]) + abs(a[1] - b[1]);
    }
};

File: problems/573/solution.py
class Solution:
    def dist(self, a, b):
        [x0, y0], [x1, y1] = a, b
        
        return abs(x0 - x1) + abs(y0 - y1)
    
    def minDistance(self, height: int, width: int, tree: List[int], squirrel: List[int], nuts: List[List[int]]) -> int:
        ans, [ta, tb], maxSavedDist = 0, tree, -20000
        
        for nut in nuts:
            ans += 2 * self.dist(nut, tree)
            maxSavedDist = max(maxSavedDist, self.dist(nut, tree) - self.dist(nut, squirrel))
        
        return ans - maxSavedDist